build/toolchain/mac/BUILD.gn

```diff
@@ -250,8 +250,22 @@ mac_toolchain("ios_clang_arm") {
   sysroot_flags = "-isysroot $ios_device_sdk_path -miphoneos-version-min=$ios_deployment_target"
 }
 
-# Toolchain used for iOS simulator targets.
-mac_toolchain("ios_clang_x64") {
+# Toolchain used for iOS simulator targets (arm64).
+mac_toolchain("ios_clang_arm_sim") {
+  toolchain_cpu = "arm"
+  toolchain_os = "mac"
+  # TODO(gw280): We have to use llvm from xcode until we upgrade our copy of llvm
```

Comment:
Does this mean Xcode 12 is required to be selected in the LUCI recipe?  I think it's on 11.6.
https://github.com/flutter/infra/blob/a39a176cf6248d008aa3534ff7c233b4beac2d0a/config/engine_config.star#L20

Reply:
I suspect so. I'm not sure what exact version of Xcode we need to be able to build on M1 but 12 seems like a reasonable assumption. 

I'd say we should target getting the machinery in place to build locally first, then we can deal with getting infra the right versions later.